### ISSUE TITLE
Wahlmöglichkeit bei AB-Ansage

### ISSUE THREAD

**Issue by Jo1972** [haumacher/phoneblock#122]
Hallo,

vorab möchte ich mich betonen, dass ich das Projekt und die Arbeit dahinter sehr schätze und keinesfalls kritisieren möchte.

Meine Frage bzw. Vorschlag geht dahin, ob es möglich gemacht werden kann, zwischen Ansagen des Spam-AB zu wählen oder gar eine eigene zu hinterlegen?

Hintergrund ist der, dass ich persönlich (und ich betone ausdrücklich persönlich) - auch wenn es sich um Spam-Anrufe handelt - es nicht als besonders professionell/seriös erachte, den Anrufer zu veräppeln, indem ich vorgaukle, dass sich eine reale Person an Apparat befindet. (Deshalb nutze ich den AB auch nicht).

Besser macht es z.B. "Frank geht ran" @ https://digitalcourage.de/frank-geht-ran

Bitte als neutralen (Verbesserungs)Vorschlag ansehen und nicht als Kritik😊

Danke und schöne Weihnachten.

**Comment by haumacher** [comment]
Ja, das ist eine gute Idee, vielen Dank für den Vorschlag! Eine Wahlmöglichkeit für die Ansage sollte nicht so schwierig sein. Eine eigene Ansage ist schon eher etwas herausvordernd - benötigt ja auch Speicherplatz auf dem Server.

**Comment by MarkyMarkDE** [comment]
Eigentlich war soetwas in der Art schon vor langer Zeit ein Wunsch von mir, während wir den AB noch im Test hatten. Inzwischen muss ich aber sagen, dass ich diesen Telefon-Paule gar nicht so schlecht und sogar irgendwie witzig finde. Zwischen unterschiedlichen Ansagen auswählen zu können, wäre ein nettes Gimmick und auch Ressourceschonend, anders als persönliche Ansagen, selbst wenn man zb. ein Limit von 50 Sekunden fixieren würde, denn viel Kleinvieh mach halt auch Mist.

**Comment by haumacher** [comment]
Mit der self-hosted Version des Anrufbeantworters kann man das natürlich sofort realisieren. Es gibt Beispiele, wo der AB auf diese Weise umgebaut wurde: https://github.com/haumacher/phoneblock/discussions/21#discussioncomment-12514939

**Comment by sristow646** [comment]
Ich möchte mal mein Beispiel oder Workflow auf das Haumacher verweist mal genauer erläutern:

Bei normalen AB wird sofort die Ansage abgespielt. Somit bekommt der eigentliche SPAMer die Ansage gar nicht mit.
Das liegt daran das meistens Call-Robots anrufen und erst wenn jemand am Telefon sich meldet, wird der Anruf an eine Person weitergeleitet. Das erkennt man daran, dass zumindest bei mir in den logs zweimal den Sound aus „waiting“ abspielt (das ist ein leises rauschen von jeweils 7 sec)

Und genau hier liegt der Verteil vom Phoneblocker AB.

1)	Also wenn Phoneblocker den Anruf annimmt, dann spielt er die Ansage aus dem Ordner „hello“ ab.
2)	jetzt wird der Sound aus dem Ordner „waiting“ abgespielt. Das so lange bis jemand sich meldet oder aufgelegt wird
3)	Dann wird die Ansage aus dem Ordner „question“ abgespielt.

 Und der Text aus dem Ordner „question“ ist dann bei mir:
/**
Guten Tag,
Sie haben eine Rufnummer angerufen, die vor Telefonspam geschützt ist.
Um den Anschlussinhaber zu schützen, wurde ich als intelligenter Anrufbeantworter zwischengeschaltet.
Die von Ihnen verwendete Nummer wird von bekannten Quellen als potenzieller Werbeanruf gemeldet.
Der Anschlussinhaber untersagt hiermit ausdrücklich jede weitere Kontaktaufnahme und fordert Sie auf, seine Kontaktdaten unverzüglich aus Ihren Systemen zu löschen.
Unerlaubte Telefonwerbung ist gemäß § 7 UWG verboten.
Sollte Ihr Anruf berechtigt sein, kontaktieren Sie den Anschlussinhaber bitte schriftlich per E-Mail oder Post.
Hinweis zum Datenschutz: Dieser Anruf wurde technisch verarbeitet, aber nicht gespeichert. Der Anschlussinhaber hat keine Kontrolle über externe Sperr- oder Spamlistsysteme.
Vielen Dank für Ihr Verständnis.
**/
Somit muss eigentlich gar nichts umgebaut werden und ihr habt den Vorteil, dass der Gegenpart die Ansage ziemlich sicher auch hört.

Es gibt da dann halt nur noch rechtliche Aspekte, die man beachten sollte:
In der Ansage gebe ich an das Aufnahmen vom Anrufer nicht gespeichert werden. (also bitte dies auch nicht tun)

Die Sound Dateien könnt ihr unter https://github.com/sristow646/conversation-sample downloaden

**Comment by haumacher** [comment]
Ein etwas anders gelagerter Wunsch, aber auch unter dem Motto Wählbarkeit der Ansage: "Gute erste Frage des AB nach dem Hallo/Hi . . . "Worum geht es?" Auswahl einer männlichen, weiblichen oder zufälligen Stimme im AB (da die Spammer oft wissen mit wem sie sprechen wollen). Mehrere verschiedene AB Stimmen, die sich abwechseln."
